libhoth_spi.c

```diff
@@ -284,6 +296,111 @@ int libhoth_spi_receive_response(struct libhoth_device* dev, void* response,
   return LIBHOTH_OK;
 }
 
+int libhoth_spi_buffer_request(struct libhoth_device* dev, const void* request,
+                               size_t request_size) {
+  if (dev == NULL) {
+    return LIBHOTH_ERR_INVALID_PARAMETER;
+  }
+
+  struct libhoth_spi_device* spi_dev =
+      (struct libhoth_spi_device*)dev->user_ctx;
+
+  if (spi_dev->buffered_request != NULL) {
+    return LIBHOTH_ERR_INTERFACE_BUSY;
+  }
+
+  spi_dev->buffered_request = malloc(request_size);
+  spi_dev->buffered_request_size = request_size;
+  memcpy(spi_dev->buffered_request, request, request_size);
+
+  return LIBHOTH_OK;
+}
+
+int libhoth_spi_send_and_receive_response(struct libhoth_device* dev,
+                                          void* response,
+                                          size_t max_response_size,
+                                          size_t* actual_size, int timeout_ms) {
+  if (dev == NULL) {
+    return LIBHOTH_ERR_INVALID_PARAMETER;
+  }
+
+  if (max_response_size < 8) {
```

Comment:
nit: replace `8` with `sizeof(struct ec_host_response)`

Reply:
Done, here and elsewhere.